app/components/primer/alpha/modal_dialog.ts

```diff
@@ -30,20 +30,17 @@ function clickHandler(event: Event) {
       return
     }
   }
-  // Find the top level dialog that is open.
-  const topLevelDialog = overlayStack[overlayStack.length - 1]
-  if (!topLevelDialog) return
```

Comment:
It did not seem like we needed to rely on checking if the "top level" dialog was open at this point. The function should only be ran the moment a user clicks the close button. This should only target the dialog where that event stemmed, and shouldn't affect any other dialogs that are open.

I removed this logic and replaced it with a conditional that will only check if one of the two acceptable button `data-*` attributes existed.